app/helpers/schedules_helper.rb

```diff
@@ -5,4 +5,10 @@ def schedule_proposal(proposal_code)
     proposal = Proposal.find_by(code: proposal_code)
     proposal.present? ? "[#{proposal.code}] #{proposal.title}" : ''
   end
+
+  def find_schedule_proposal(proposal_code)
+    return '' if proposal_code.blank?
+
+    Proposal.find_by(code: proposal_code)
+  end
```

Comment:
You can just use [Proposal.find(proposal_code](https://github.com/birs-math/proposals/blob/000223c1c06f53caf917304caa3a80232a034497/app/models/proposal.rb#L137).